File: cocktails/api/CocktailApi.js
import axios from 'axios';

const CocktailApi = axios.create({
    baseURL: 'http://www.thecocktaildb.com/api/json/v1/1/'
});

export const Filter = type => {
    return CocktailApi.get('filter.php?g=' + type)
        .then(response => response.data.drinks)
        .catch(error => { console.log(error) })
        ;
};

export const GetDetails = id => CocktailApi.get('/lookup.php?i=' + id)
.then(response => response.data 
                    && response.data.drinks
                    ? response.data.drinks[0]
                    : null);

File: MyApplication/app/src/main/java/com/buoy/codetest/model/api/CocktailApi.kt
package com.buoy.codetest.model.api

import com.buoy.codetest.model.domain.DrinksList
import io.reactivex.Single

import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface CocktailApi {

    @GET("api/json/v1/1/filter.php?g=Cocktail_glass")
    fun getAllCocktails(): Single<DrinksList>

    @GET("api/json/v1/1/lookup.php")
    fun getCocktailDetail(@Query("i") id: String): Single<DrinksList>

    companion object {
        const val URL = "http://www.thecocktaildb.com"
    }
}